src/io/iohk/scalanet/peergroup/SimplePeerGroup.scala

```diff
@@ -37,28 +38,39 @@ class SimplePeerGroup[A, AA](
     .messageChannel[EnrolMe[A, AA]]
     .foreach {
       case (_, enrolMe) =>
-        routingTable += enrolMe.myAddress -> enrolMe.myUnderlyingAddress
+        enrolMe.multiCastAddresses foreach { a =>
```

Comment:
this is becoming a bit verbose. can you pull out one or more private methods that describes what is happening?